stonesoup/sensor/radar/radar.py

```diff
@@ -0,0 +1,453 @@
+                               " of the target will change for each timestep. "
+                               "The random RCS follows the probability "
+                               "distribution of the Swerling 1 case.")
+    rcs = Property(float, default=None, doc="The radar cross section of "
+                                            "targets in meters squared.")
+
+    probability_false_alarm = Property(Probability, default=1e-6,
+                                       doc="Probability of false alarm used in"
+                                           " the North's approximation")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        temp = 290  # noise reference temperature (room temperature kelvin)
+        # convert from dB
+        noise_figure = 10 ** (self.receiver_noise / 10)
+        loss = 10 ** (self.loss / 10)
+        # calculate part of snr that is independent of:
+        #   rcs, transmitted power, gain and range
+        self.snr_constant = (const.c ** 2 * self.number_pulses *
+                             self.duty_cycle) / (64 * np.pi ** 3 * const.k *
+                                                 temp * self.band_width *
+                                                 noise_figure * self.frequency
+                                                 ** 2 * loss)
+        # calculate range resolution for compressed pulse
+
+    def swerling_1(self, rcs):
+        r"""Swerling 1 case
+
+         Returns
+        -------
+        :class:`float`
+        returns random RCS based on the Swerling 1 case
+
+        .. math::
+
+            P\left(\sigma\right) = \dfrac{1}{\sigma_{average}} \exp{\left(
+            \dfrac{-\sigma}{\sigma_{average}}\right)}
+
+        """
+        return -rcs * np.log(np.random.rand())
+
+    @property
+    def _rotation_matrix(self):
+        """_rotation_matrix getter method
+
+        Calculates and returns the (3D) axis rotation matrix.
+
+        Returns
+        -------
+        :class:`numpy.ndarray` of shape (3, 3)
+            The model (3D) rotation matrix.
+        """
+
+        theta_x = -self.rotation_offset[0, 0]  # roll
+        theta_y = -self.rotation_offset[1, 0]  # pitch#elevation
+        theta_z = -self.rotation_offset[2, 0]  # yaw#azimuth
+
+        return rotz(theta_z) @ roty(theta_y) @ rotx(theta_x)
+
+    def gen_probability(self, sky_state):
+        """Generates probability of detection of a given State.
+
+        Parameters
+        ----------
+        sky_state : The target state.
+
+        Returns
+        -------
+        det_prob : Probability of detection.
+        snr : Signal to noise ratio.
+        rcs : RCS after the Swerling 1 case is applied.
+        directed_power : Power in the direction of the target.
+        spoiled_gain : Gain in decibels with beam spoiling applied.
+        spoiled_width : Beam width with beam spoiling applied.
```

Comment:
Syntax of these is wrong, as after colon should be type, and description on line below. See [5. Returns](https://numpydoc.readthedocs.io/en/latest/format.html#sections).